### PR TITLE
Add organization ID to DevPortal rest APIs

### PR DESCRIPTION
**Pull request by YasasRangika**
### Goals
To reduce the customization of master branch done on top of Choreo for supporting the organization concept.

### Approach
This PR adds and refactors the following changes:
- Adds persistence layer changes to deleteAPI method in publisher rest api
- Adds organization ID to getAllCommentsOfAPI method in DevPortal rest api.
- Adds organization ID to addCommentToAPI method in DevPortal rest api.
- Removes unused tenant ID from apisApiIdRatingsGet, apisApiIdUserRatingGet, apisApiIdUserRatingPut, apisApiIdUserRatingDelete methods in DevPortal rest api.
- Adds organization ID to apisApiIdDocumentsGet method in DevPortal rest api.
- Adds organization ID to apisApiIdDocumentsDocumentIdContentGet method in DevPortal rest api.

https://docs.google.com/spreadsheets/d/18GA4odAnRz2cB0Ris3RL_62_yDfE3Jb52j5LP5BI2OI/edit#gid=1530511003

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java [hunk 1]
```diff
@@ -881,6 +881,7 @@
             "   AND API.API_ID = SUBS.API_ID " +
             "   AND API.API_NAME = ? " +
             "   AND API.API_VERSION = ? " +
+            "   AND API.ORGANIZATION_ID= ? " +
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java:884]: This should be organization

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java [hunk 2]
```diff
@@ -846,24 +820,10 @@ public Response apisApiIdTopicsGet(String apiId, String xWSO2Tenant, MessageCont
...
         try {
             int rating = 0;
             String username = RestApiCommonUtil.getLoggedInUsername();
-            if (!APIUtil.isTenantAvailable(requestedTenantDomain)) {
-                RestApiUtil.handleBadRequest("Provided tenant domain '" + xWSO2Tenant + "' is invalid",
-                        ExceptionCodes.INVALID_TENANT.getErrorCode(), log);
-            }
             APIConsumer apiConsumer = RestApiCommonUtil.getConsumer(username);
-            //this will fail if user doesn't have access to the API or the API does not exist
-            ApiTypeWrapper apiTypeWrapper = apiConsumer.getAPIorAPIProductByUUID(id, requestedTenantDomain);
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java:859]: We will hv to keep this check for api visibility related checks. Without this, users who do not hv visibility permission can edit it. Check wether it is handled from different place

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java [hunk 3]
```diff
@@ -951,33 +903,16 @@ public Response apisApiIdUserRatingGet(String id, String xWSO2Tenant, String ifN
...
-        String requestedTenantDomain = RestApiUtil.getRequestedTenantDomain(xWSO2Tenant);
         try {
             String username = RestApiCommonUtil.getLoggedInUsername();
-            if (!APIUtil.isTenantAvailable(requestedTenantDomain)) {
-                RestApiUtil.handleBadRequest("Provided tenant domain '" + xWSO2Tenant + "' is invalid",
-                        ExceptionCodes.INVALID_TENANT.getErrorCode(), log);
-            }
             APIConsumer apiConsumer = RestApiCommonUtil.getConsumer(username);
-            //this will fail if user doesn't have access to the API or the API does not exist
-            ApiTypeWrapper apiTypeWrapper = apiConsumer.getAPIorAPIProductByUUID(apiId, requestedTenantDomain);
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java:973]: Same here

**Review by chamilaadhi — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java (hunk 2 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java:859]: we could add a similar check done in https://github.com/wso2/carbon-apimgt/blob/master-org-refactor/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java#L6117 here as well

Comment (YasasRangika) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java:859]: fixed with https://github.com/wso2/carbon-apimgt/pull/10512/commits/2bbd75f03d2e35bf306fa99a5e37c4268b9e3b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java (hunk 3 above)
Comment (YasasRangika) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java:973]: fixed with https://github.com/wso2/carbon-apimgt/pull/10512/commits/2bbd75f03d2e35bf306fa99a5e37c4268b9e3b9a

**Review by chamilaadhi — APPROVED** [review]

**Pull request merged**